org.scala-ide.sdt.core/src/org/scalaide/ui/internal/preferences/StatisticsPreferencePage.scala

```diff
@@ -0,0 +1,177 @@
+package org.scalaide.ui.internal.preferences
+
+import java.text.DateFormat
+import java.util.Calendar
+
+import org.eclipse.jface.layout.TableColumnLayout
+import org.eclipse.jface.preference.PreferencePage
+import org.eclipse.jface.viewers.ColumnWeightData
+import org.eclipse.jface.viewers.IStructuredContentProvider
+import org.eclipse.jface.viewers.TableViewer
+import org.eclipse.jface.viewers.TableViewerColumn
+import org.eclipse.jface.viewers.Viewer
+import org.eclipse.jface.viewers.ViewerSorter
+import org.eclipse.swt.SWT
+import org.eclipse.swt.events.SelectionEvent
+import org.eclipse.swt.layout.GridData
+import org.eclipse.swt.layout.GridLayout
+import org.eclipse.swt.widgets.Composite
+import org.eclipse.swt.widgets.Control
+import org.eclipse.swt.widgets.Table
+import org.eclipse.ui.IWorkbench
+import org.eclipse.ui.IWorkbenchPreferencePage
+import org.scalaide.core.internal.ScalaPlugin
+import org.scalaide.core.internal.statistics.FeatureData
+import org.scalaide.core.internal.statistics.Features.CharactersSaved
+import org.scalaide.core.internal.statistics.Features.NotSpecified
+import org.scalaide.util.eclipse.SWTUtils._
+
+class StatisticsPreferencePage extends PreferencePage with IWorkbenchPreferencePage {
+
+  /*
+   * Some features should not be displayed in the list of used features. These
+   * are part of `filteredData`, the displayed ones are `data`
+   */
+  private val (data, filteredData) = {
+    val d = ScalaPlugin().statistics.data.toArray
+    d.partition(fd ⇒ !Set(CharactersSaved, NotSpecified)(fd.feature))
+  }
+
+  def createContents(parent: Composite): Control = {
+    val base = new Composite(parent, SWT.NONE)
+    base.setLayout(new GridLayout(1, true))
+
+    val charsSaved = filteredData.find(_.feature == CharactersSaved).get
```

Comment:
Is it possible that someone didn't used code completion at all (so there is no characters saved) and that `get` will be called resulting in exception?